glambie/data/timeseries_data.py

```diff
@@ -26,21 +30,42 @@ def max_change_value(self) -> float:
         finite_list = np.isfinite(self.changes)
         return np.max(self.changes[finite_list])
 
-    def __init__(self, dates: np.ndarray, area: np.ndarray,
-                 changes: np.ndarray, errors: np.ndarray, rgi_version: int, region_id: int, unit: str,
-                 user: Optional[str] = None, user_group: Optional[str] = None, data_group: Optional[str] = None):
+    def __init__(self, data_group: Optional[GlambieDataGroup] = None, region: Optional[RGIRegion] = None,
```

Comment:
We could reuse the dataclass approach here. I had to play with it myself but for optional values you just use something like 
```
from dataclasses import dataclass


@dataclass
class ClassWithDefaults():
    mask: str
    msg: str = None

    def test(self):
        return(self.msg)


cls = ClassWithDefaults(mask="str")

print(cls.mask, cls.msg, cls.test())
```